### PR TITLE
fix(privates-database): improve configuration management

### PR DESCRIPTION
**Pull request by wdhif**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix PUDBLOW-847
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [X] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [x] Breaking change is mentioned in relevant commits

## Description

Improve configuration management and add sql_modes special case.

### CODE REVIEW

**Comment by wdhif** [comment]
image

**Review by Meuh-42 — DISMISSED** [review]

**Review by mferon — APPROVED** [review]

**Review by Meuh-42 — APPROVED** [review]

File: packages/manager/apps/web/client/app/private-database/configuration/private-database-configuration.controller.js [hunk 1]
```diff
@@ -62,25 +62,34 @@ angular.module('App').controller(
         id: value,
         text: value,
       }));
-      if (field.key === 'autocommit') {
-        PrivateDatabaseConfigurationsCtrl.convertFieldAsToggle(field);
-      } else if (field.key === 'event_scheduler') {
-        PrivateDatabaseConfigurationsCtrl.convertFieldAsToggle(field);
+
+      if (field.key === 'sql_mode') {
```
Comment (radireddy) [inline packages/manager/apps/web/client/app/private-database/configuration/private-database-configuration.controller.js:66]: NITPICK: Not a blocker
Is it possible to move field.key, field.type and other constants into a separate constant file? or create a ENUM?

**Review by radireddy — APPROVED** [review]

File: packages/manager/apps/web/client/app/private-database/configuration/private-database-configuration.controller.js (hunk 1 above)
Comment (wdhif) [reply packages/manager/apps/web/client/app/private-database/configuration/private-database-configuration.controller.js:66]: This code is pretty old and should be reworked at some point. But right now we don't want to do that.

**Comment by Meuh-42** [comment]
Hi, I confirm that staging is valid

**Review by mohammed-zahaf — APPROVED** [review]

**Pull request merged**